### PR TITLE
Docs: add config concepts page with python config info

### PR DESCRIPTION
**Pull request by treysp**
This PR does two things:

1. Separates exposition and reference information into separate pages (new guide page for exposition)
2. Adds documentation for Python config files

Note: the first ~100 lines of the guide is new Python content, the rest is copied over from the reference page with slight modifications.

### CODE REVIEW

File: docs/concepts/configuration/overview.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+```
+
+#### Engine connection configuration
+
+These pages describe the connection configuration options for each execution engine.
+
+* [BigQuery](../../integrations/engines/bigquery.md)
+* [Databricks](../../integrations/engines/databricks.md)
+* [DuckDB](../../integrations/engines/duckdb.md)
+* [MySQL](../../integrations/engines/mysql.md)
```
Comment (izeigerman) [inline docs/concepts/configuration/overview.md:218]: Just a heads up that the MSSQL link has been added to this list recently.

File: sqlmesh/core/config/root.py [hunk 2]
```diff
@@ -40,9 +39,13 @@ class Config(BaseConfig):
         auto_categorize_changes: Indicates whether SQLMesh should attempt to automatically categorize model changes (breaking / non-breaking)
             during plan creation.
         users: A list of users that can be used for approvals/notifications.
+        username: Name of a single user who should receive approvals/notification, instead of all users in the `users` list.
         pinned_environments: A list of development environment names that should not be deleted by the janitor task.
+        loader: Loader class used create a SQLMesh context from project files.
```
Comment (izeigerman) [inline sqlmesh/core/config/root.py:44]:
```suggestion
        loader: Loader class used for loading project files.
```

File: docs/reference/configuration.md [hunk 3]
```diff
@@ -46,103 +29,36 @@ Configuration options for SQLMesh environment creation and promotion.
...
 | `default_target_environment` | The name of the environment that will be the default target for the `sqlmesh plan` and `sqlmesh run` commands. (Default: `prod`)                                                                                                                                                                   | string               | N        |
-| `physical_schema_override`   | The name of the schema where SQLMesh should place physical tables.
-
-### Virtual Data Environments
-
-Configuration options for SQLMesh virtual data environment schemas.
-
-#### Physical Schema Override
-By default, SQLMesh creates physical tables for a model with a naming convention of `sqlmesh__[model schema]`.
+| `physical_schema_override`   | The name of the schema where SQLMesh should place physical tables - [addition details](../guides/configuration.md#physical-schema-override). (Default: `sqlmesh__[model schema]`)                                                                                                                                                                   | string               | N        |
```
Comment (izeigerman) [inline docs/reference/configuration.md:34]: This is not "name" though, it's a mapping.

File: sqlmesh/core/config/root.py [hunk 4]
```diff
@@ -40,9 +39,13 @@ class Config(BaseConfig):
...
+        env_vars: A dictionary of environmental variable names and values.
         model_defaults: Default values for model definitions.
         include_unmodified: Indicates whether to include unmodified models in the target development environment.
+        physical_schema_override: A dictionary of model schema names and the physical schemas SQLMesh should use for the models' physical tables.
```
Comment (izeigerman) [inline sqlmesh/core/config/root.py:48]: "A mapping from model schema names to names of schemas in which physical tables for corresponding models will be placed"

**Review by eakmanrq — APPROVED** [review]

**Review by izeigerman — APPROVED** [review]

**Pull request merged**